docs/langref/relay/intro.rst

```diff
@@ -0,0 +1,205 @@
+can write a global identifier as ::@global.
+
+Local Variable
+~~~~~~~~~~~~~~~~~
+
+Local identifiers are prefixed by the :code:`%` sigil. A local identifier always
+references a parameter, or let bound expression. You can write a local
+identifier as :code:`%local`.
+
+
+================
+Global Functions
+================
+
+A definition consists of a name, type parameter, parameters, and an optional return
+type. A Relay definition is similar to a procedure or function in a typical programming
+language, but can also be viewed as a named subgraph.
+
+A definition minimally consists of an identifier :code:`@id`, an empty set of
+parameters, and a body expression contained by curly braces
+
+.. code-block:: python
+
+    def @id() { body }
+
+A definiton may also contain any number of parameters, for example a
+simple function which just adds two tensors
+
+.. code-block:: python
+
+    def @add(%x, %y) { %x + %y }
+
+It is also possible for us to annotate explicit types on definitions, for example
+we can restrict the above definition to only work on certain types
+
+.. code-block:: python
+
+    def @add(%x: Tensor<Float,10, 10>, %y: Tensor<Float, 10, 10>) -> Tensor<Float, 10, 10> {
+        %x + %y
+    }
+
+A parameter is just a pairing of a :py:class:`~tvm.relay.expr.LocalVar` and optional :py:class:`~tvm.relay.type.Type`. They represent
+the formal parameters of functions and definitions, and are written as :code:`%x : T`.
+
+They may only appear in function literals, and definitions, and have no relation
+to parameters in the machine learning.
+
+When the type information is omitted we will attempt to infer a most general type
+for the users. This property is known as generalization, for a definition without
+explicit annotations, we will attempt to assign the most general type. When the
+return type is omitted we will infer the return type based on the text of the
+program.
+
+Finally we can directly construct type polymorphic definitions by writing down
+a set of type parameters for a definition. To define a polymoprhic identity
+function, the function which just returns its argument as so.
+::
+    def @id<s: Shape, bt: BaseType>(%x: Tensor<bt, s>) {
+        %x
+    }
+
+Notice we can omit the return type, and it will still be inferred.
+
+*Note: this is not yet implemented.*
+
+Finally we allo a definition be prefixed by metadata, which adds
+extra properties to the definition.
+
+It is important to be able to annotate metadata that is external to
+the computational behavior of a definition. For example we can use
+this to add an `inline` or `noinline` attribute which the compiler
+can consider when performing inlining.
+
+For example we can set the attributes for :code:`@id_real`.::
+
+
+    attributes id_real {
+        inline: true
+    }
+
+    def id_real(%x:Real) { ret %x }
+
+
+=========
+Operators
+=========
+
+A primitive operation that is not defined in the Relay language but provided
```

Comment:
The first sentence here is a fragment. Perhaps it should read, "An operator is a primitive operation that is not defined in the Relay language but is provided externally"?